python/cudaq/__init__.py

```diff
@@ -73,7 +74,8 @@
 Target = cudaq_runtime.Target
 State = cudaq_runtime.State
 pauli_word = cudaq_runtime.pauli_word
-Tensor = cudaq_runtime.Tensor
+# FIXME What to do with this Tensor, is it used publicly?
+StateTensor = cudaq_runtime.Tensor
```

Comment:
@annagrin ?

Reply:
@1tnguyen do you think we can remove it from here?

Reply:
The type was exposed before so technically this change can have effect on the user code, for example:

the following user code
```python 
def foo(t: list[cudaq.Tensor]):
        return t
```
would need to be changed to
```python
def foo(t: list[cudaq.StateTensor]):
        return t
```

I would think that this is a rare (and may be non-existent) occurrence, but I am not really familiar with the user code and our process on changing language... 

@efratshabtai do you know if we can just rename an already existing type if there is low chance of it being already used?

Reply:
This should not be exposed at the CUDA-Q kernel level. If you are talking about annotations for CUDA-Q kernels, the type to be used there should just be cudaq.State. Why would we allow any general Tensor as input to a kernel? And if the above are just regular Python functions, why annotate those?

Reply:
Is it referenced/used anywhere? If not, let's just get rid of it.